src/pages/_app.tsx

```diff
@@ -1,6 +1,25 @@
 import '@/styles/globals.css'
 import type { AppProps } from 'next/app'
+import { useState } from 'react';
+import { useRouter } from 'next/router';
 
 export default function App({ Component, pageProps }: AppProps) {
-  return <Component {...pageProps} />
+  const [user, setUser] = useState<number | null>(null)
```

Comment:
I'm thinking that using context could be a great way to keep track of our user without having to pass these props to every component